src/entities-manager.ts

```diff
@@ -0,0 +1,44 @@
+import { Subject } from "joyst";
+
+export type Entity = {
+    id: number;
+    x: number;
+    y: number;
+};
+
+export class EntitiesManager {
+    public static entities: Subject<Entity>[] = [];
+
+    public static entityIds = new Subject<number[]>([]);
+
+    private static nextId = 0;
+
+    public static add(): Subject<Entity> {
+        const newId = this.nextId++;
+
+        const newEntitySubject = new Subject({
+            x: 0,
+            y: 0,
+            id: newId
+        });
```

Comment:
Should entity initialization be decoupled from the `add` function, or will added entities always be this default object? Would it be better to have separate `create` and `add` methods and add entities through dependency injection?

Reply:
I could definitely see it being worth decoupling entities/platforms from the `add`  method especially since both managers are functionally pretty identical. This will be especially true if/when we add an initialization modal so that a new entity isn't created with a default values.